test/e2e/TokenGatedRoleAuthorizerE2E.t.sol

```diff
@@ -0,0 +1,219 @@
+        }
+        vm.stopPrank();
+
+        //--------------------------------------------------------------------------------
+        // Set up seed deposit and initial deposit by users
+        //--------------------------------------------------------------------------------
+
+        uint initialDeposit = 10e18;
+        token.mint(address(this), initialDeposit);
+        token.approve(address(fundingManager), initialDeposit);
+        fundingManager.deposit(initialDeposit);
+
+        address funder1 = makeAddr("funder1");
+
+        token.mint(funder1, 1000e18);
+
+        vm.startPrank(funder1);
+        {
+            token.approve(address(fundingManager), 1000e18);
+            fundingManager.deposit(1000e18);
+        }
+        vm.stopPrank();
+
+        //--------------------------------------------------------------------------------
+        // Create bounty
+        //--------------------------------------------------------------------------------
+        vm.prank(orchestratorOwner);
+        uint bountyId =
+            bountyManager.addBounty(100e18, 500e18, "This is a test bounty");
+
+        // others can't add it
+        vm.expectRevert();
+        vm.prank(orchestratorManager);
+        bountyManager.addBounty(100e18, 500e18, "This is a test bounty");
+
+        vm.expectRevert();
+        vm.prank(bountySubmitter);
+        bountyManager.addBounty(100e18, 500e18, "This is a test bounty");
+
+        // Validate
+        IBountyManager.Bounty memory bounty =
+            bountyManager.getBountyInformation(1);
+        assertEq(bounty.minimumPayoutAmount, 100e18);
+        assertEq(bounty.maximumPayoutAmount, 500e18);
+        assertEq(bounty.details, "This is a test bounty");
+
+        //--------------------------------------------------------------------------------
+        // Worker submits bounty
+        //--------------------------------------------------------------------------------
+        vm.startPrank(bountySubmitter);
+        IBountyManager.Contributor memory BOB =
+            IBountyManager.Contributor(bountySubmitter, 200e18);
+
+        IBountyManager.Contributor[] memory contribs =
+            new IBountyManager.Contributor[](1);
+        contribs[0] = BOB;
+
+        uint claimId = bountyManager.addClaim(
+            bountyId, contribs, "This is a test submission"
+        );
+
+        vm.stopPrank();
+
+        //--------------------------------------------------------------------------------
+        // Manager verifies bounty claim
+        //--------------------------------------------------------------------------------
+        vm.prank(orchestratorManager);
+        bountyManager.verifyClaim(claimId, contribs);
+
+        // the worker can't verifiy it
+        vm.expectRevert();
+        vm.prank(bountySubmitter);
+        bountyManager.addBounty(100e18, 500e18, "This is a test bounty");
+    }
+
+    function _createNewOrchestratorWithAllModules_withTokenGatedRoleAuthorizer(
```

Comment:
Lol this here is a great idea
Why didnt we use this in the other e2e tests?
Basically building the Orchestrator directly in here instead of the base e2e test
I always felt like it was pretty ugly to see in the base e2e test

Reply:
Yeah, maybe we should refactor the other e2e into that format